scripts/run_tracker.py

```diff
@@ -172,6 +178,156 @@ def trigger_analysis(run, config):
             LOG.warn(("Something went wrong when triggering the analysis of {}. Please "
                       "check the logfile and make sure to start the analysis!".format(os.path.basename(run))))
 
+def prepare_sample_sheet(run, config):
+    """ This is a temporary function in order to solve the current problem with LIMS system
+        not able to generate a compatible samplesheet for HiSeqX. This function needs to massage 
+        the sample sheet created by GenoLogics in order to correctly demultiplex HiSeqX runs.
+        This function returns with success if the samplesheet is in the correct place, otherwise
+        this flowcell will not be processed.
+        
+        :param str run: Run directory
+        :param dict config: Parset configuration file
+        
+    """
+    #start by checking if samplesheet is in the correct place
+    run_name     = os.path.basename(run)
+    current_year = '20' + run_name[0:2]
+    samplesheets_dirs = config['samplesheets_dir']
+    samplesheets_dir  = os.path.join(samplesheets_dirs, current_year)
+
+    run_name_componets = run_name.split("_")
+    FCID = run_name_componets[3][1:]
+    
+    FCID_samplesheet_origin = os.path.join(samplesheets_dir, FCID + '.csv')
+    FCID_samplesheet_dest   = os.path.join(run, "SampleSheet.csv")
+
+    #check that the samplesheet is not already present
+    if os.path.exists(FCID_samplesheet_dest):
+        LOG.warn(("When trying to generate SampleSheet.csv for sample sheet {} I dinf out that "
+                 "SampleSheet.csv was already present in {} !!".format(FCID, FCID_samplesheet_dest)))
+        return False
+
+    FCID_samplesheet_origin_dict = sampleSheetToDict(FCID_samplesheet_origin) # store samplesheet in a dict
+
+    FCID_samplesheet_dest_dict = sampleSheetMassage(FCID_samplesheet_origin_dict) # massage the sample sheet
+
+    if not FCID_samplesheet_dest_dict: # something went wrong when working on samplesheet
+        return False
+
+    dictToSampleSheet(FCID_samplesheet_dest_dict, FCID_samplesheet_dest) # store the new smaplesheet file in the fc dir
+
+    return True #everything ended corretly
+
+
+def sampleSheetMassage(samplesheet_dict):
```

Comment:
ok that is just picky but... the rest of the script (and generally our Python code) is not using CamelCase for naming functions.